site/app/controllers/forum/ForumController.php

```diff
@@ -1363,57 +1364,62 @@ public function getSplitPostInfo() {
      */
     #[Route("/courses/{_semester}/{_course}/forum/posts/history", methods: ["POST"])]
     public function getHistory() {
+        $repo = $this->core->getCourseEntityManager()->getRepository(Post::class);
         $post_id = $_POST["post_id"];
         $output = [];
-        $_post = [];
-        $older_posts = $this->core->getQueries()->getPostHistory($post_id);
-        $current_post = $this->core->getQueries()->getPost($post_id);
-        $post_attachments = $this->core->getQueries()->getForumAttachments([$post_id], true);
-        $oc = $current_post["author_user_id"];
-        $anon = $current_post["anonymous"];
+        $post = $repo->getPostWithHistory($post_id);
+
         $GLOBALS['totalAttachments'] = 0;
         $edit_id = 0;
-        foreach ($older_posts as $post) {
-            $_post['user'] = !$this->modifyAnonymous($oc) && $oc == $post["edit_author"] && $anon ? '' : $post["edit_author"];
-            $_post['content'] = $this->core->getOutput()->renderTwigTemplate("forum/RenderPost.twig", [
-                "post_content" => $post["content"],
+        foreach ($post->getHistory() as $version) {
+            $tmp = [];
+            $tmp['user'] = (!$this->modifyAnonymous($post->getAuthor()->getId()) && $post->getAuthor()->getId() == $version->getEditAuthor()->getId() && $post->isAnonymous()) ? '' : $version->getEditAuthor()->getId();
```

Comment:
Can you either break this into more concise parts or add a comment describing what each condition does?